### PR TITLE
Created GitHub Action that discards closed issue channels

### LINKED ISSUE #173: Create a GitHub Action that discards closed issue channels
### What's your idea?

Building on top of issue #162, let's create a complementary NodeJS script that takes care of moving the issue channel beneath the `closed-issues-below` channel.

### PR DESCRIPTION
**Pull request by EthanThatOneKid**
When an issue is closed or deleted, the associated channel on our Discord server must be properly archived. Without automatic archival, the [channel list](https://gist.github.com/diamondburned/696e94885db5ffa29fbfc08a5f617c58) can build up very quickly. I propose that when we close an issue, the associated text channel is converted into a TXT file and sent to our main text channel.

**Note**: There is no more need to sort the text channels now that they will simply be appended and deleted atomically.

Attempting to resolve #173.

### CODE REVIEW

File: scripts/close-issue-channel/common.js [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
-		const hubChannel = channels.cache.find((c) => c.position === 0);
-		const closedChannel = channels.cache.find((c) => c.name === `website-issue-${issue.number}`);
-		// TODO: Check if hub channel is always at position 0.
-		console.log({ hubChannel, closedChannel, channels: [...channels.cache] });
-		// TODO: Fetch all messages from `closedChannel` text channel.
-		// TODO: Print messages out to TXT file format.
-		// TODO: Send TXT file to `hubChannel` text channel.
-		// TODO: Delete `closedChannel` text channel.
+		const hubChannel = channels.cache.find(
+			(c) => c.parentId === process.env.HUB_ID && c.position === 0
```
Comment (diamondburned) [inline scripts/close-issue-channel/common.js]: The `HUB_ID` should be matching the parent category? So the code should've been
`c.id == process.env.HUB_ID`, unless the `hubChannel` variable is pointing to a
channel inside the category, but that's nto a good idea.

File: scripts/close-issue-channel/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Close Issue Channel Script
+
+## Command Line Usage
+
+1. Make sure your environment variables are defined.
+1. Run the following command.
+
+```bash
+node scripts/close-issue-channel --payload="{ \"number\": 180 }"
```
Comment (diamondburned) [inline scripts/close-issue-channel/README.md]: Why not just do `--issue 180` and pass in `--issue ${{ github.event.issue.number }}`?

Comment (EthanThatOneKid) [reply scripts/close-issue-channel/README.md]: Nice catch. The issue number is all that's needed for now.

File: scripts/common.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { Client, Intents } from 'discord.js';
+import { config } from 'dotenv';
+import minimist from 'minimist';
+
+export const parseArgs = () => minimist(process.argv.slice(2));
```
Comment (diamondburned) [inline scripts/common.js]: What is `minimist` used for?

Comment (EthanThatOneKid) [reply scripts/common.js]: Minimist is just a small library that parses the CLI arguments for us. 

```
$ node example/parse.js -x 3 -y 4 -n5 -abc --beep=boop foo bar baz
{ _: [ 'foo', 'bar', 'baz' ],
  x: 3,
  y: 4,
  n: 5,
  a: true,
  b: true,
  c: true,
  beep: 'boop' }
```

https://github.com/substack/minimist

Comment (diamondburned) [reply scripts/common.js]: I don't think we need a whole library for this. We only have a single flag; the
flag could just be `$1` instead:

```sh
node script.js '{"issue": 30}'
```

```js
const payload = JSON.parse(process.argv[1])
```

Comment (EthanThatOneKid) [reply scripts/common.js]: Good idea :) resolved by <https://github.com/EthanThatOneKid/acmcsuf.com/pull/190/commits/8e78797a78e82e58ddc514f7af0633e04ed748e8>.

File: scripts/close-issue-channel.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+			day: '2-digit',
+			month: 'short',
+			year: '2-digit',
+			hour: '2-digit',
+			minute: '2-digit',
+			timeZone: 'America/Los_Angeles',
+		})
+		.replace(/\,/g, '')
+		.split(' ');
+	return `${day} ${month} ${year} ${time} ${amPm} PST`;
```
Comment (diamondburned) [inline scripts/close-issue-channel.js]: Have we forgotten about Daylight Saving Time? 😛

File: scripts/common.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { Client, Intents } from 'discord.js';
+import { config } from 'dotenv';
+
+export const get$1 = () => {
```
Comment (diamondburned) [inline scripts/common.js]: Nitpick: I think it'd be better to just name this `getNArg(n)` and use it as
`getNArg(1)` than naming it `$1`.

**Review by diamondburned — CHANGES_REQUESTED** [review]

File: scripts/close-issue-channel.js (hunk 4 above)
Comment (EthanThatOneKid) [reply scripts/close-issue-channel.js]: Should I do anything in the formatter to account for daylight saving time?

Comment (diamondburned) [reply scripts/close-issue-channel.js]: I think a function to get the short timezone name should work.

I don't think JS dates have timezones saved in them, though, so there's a slight
chance that everything will seem off a few seconds during the DST shift, but
that's only a yearly thing.

Comment (EthanThatOneKid) [reply scripts/close-issue-channel.js]: Thanks for the suggestion. Resolved by <https://github.com/EthanThatOneKid/acmcsuf.com/pull/190/commits/4d96758c460d20495ee141102c0e493b6c587d2a>.

File: scripts/common.js (hunk 5 above)
Comment (EthanThatOneKid) [reply scripts/common.js]: Cool idea. Resolved by <https://github.com/EthanThatOneKid/acmcsuf.com/pull/190/commits/14e74d093af1c9acac0da1f31d4946dea6e5925c>.

**Review by diamondburned — APPROVED** [review]

**Comment by EthanThatOneKid** [comment]
Screenshot of `website-issue-151.txt` generated by [`/.github/workflows/close_issue_channel.yaml`](https://github.com/EthanThatOneKid/acmcsuf.com/blob/main/.github/workflows/close_issue_channel.yaml) with only 4 messages.

image

**Pull request merged**